cpp/daal/src/algorithms/dtrees/forest/classification/oneapi/df_classification_train_hist_oneapi_impl.i

```diff
@@ -253,6 +254,7 @@ services::Status ClassificationTrainBatchKernelOneAPI<algorithmFPType, hist>::co
 
     auto & context = services::Environment::getInstance()->getDefaultExecutionContext();
 
+    // no overflow check is required because of _nNodesGroups and _nodeGroupProps are small constants
     auto nodesGroups = context.allocate(TypeIds::id<int>(), _nNodesGroups * _nodeGroupProps, &status);
```

Comment:
Separate status check for nodesGroups allocation?

Reply:
fixed